admin_pages/messages/Messages_Admin_Page.core.php

```diff
@@ -4018,7 +4018,7 @@ protected function _activate_messenger($messenger_name)
             : array();
 
         // ensure is active
-        $this->_message_resource_manager->activate_messenger($messenger_name, $message_types_to_activate);
+        $this->_message_resource_manager->activate_messenger($active_messenger, $message_types_to_activate);
```

Comment:
first thing `activate_messenger()` does... is retrieve the messenger object given the supplied messenger name... or ya know... you could just pass the messenger existing object you already have!!!

which... also doesn't make sense... because how come the messenger object already exists if it hasn't been activated yet?!?!?

Anyways, in like 5 out of the 6 places that `activate_messenger()` is called... the messenger object was already instantiated.